account_cash_discount_base/models/account_invoice.py

```diff
@@ -98,6 +98,14 @@ def _compute_has_discount(self):
                 rec.type == 'in_invoice'
             )
 
+    @api.model
+    def _get_discount_base_date(self):
+        discount_base_date = datetime.today()
+        if self.date_invoice:
+            discount_base_date = \
+                fields.Date.from_string(self.date_invoice)
```

Comment:
If we keep this method, it must be changed to an @api.mutli with a self.ensure_one()